src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -434,33 +434,33 @@ private DeclarationModifiers MakeModifiers(TypeKind typeKind, BindingDiagnosticB
                 }
             }
 
-            if (this.Name == SyntaxFacts.GetText(SyntaxKind.RecordKeyword))
-            {
-                foreach (var syntaxRef in SyntaxReferences)
-                {
-                    SyntaxToken? identifier = syntaxRef.GetSyntax() switch
-                    {
-                        BaseTypeDeclarationSyntax typeDecl => typeDecl.Identifier,
-                        DelegateDeclarationSyntax delegateDecl => delegateDecl.Identifier,
-                        _ => null
-                    };
-
-                    ReportTypeNamedRecord(identifier?.Text, this.DeclaringCompilation, diagnostics.DiagnosticBag, identifier?.GetLocation() ?? Location.None);
-                }
-            }
-
             return result;
         }
 
-        internal static void ReportTypeNamedRecord(string? name, CSharpCompilation compilation, DiagnosticBag? diagnostics, Location location)
+        internal static bool IsReservedTypeName(string? name)
+        {
+            return name is { Length: > 0 } && name.All(c => char.IsLower(c) && c < 128);
```

Comment:
I don't see a benefit (prefer readability over premature optimization). If anything, I wanted to use `char.IsAscii` instead of doing a lower-level check, but surprisingly it was introduced in .NET 6.

Reply:
I specifically asked because I find the current code less clear than a simple character comparison. What is 128? I don't have the ascii table memorized. Far easier to avoid the confusion and simply say "is it between 'a' and 'z'?"

Reply:
I don't think it's simpler. I can extract an `isAscii` local function (since `char.IsAscii` isn't available) if you'd like. Let me know.

Reply:
I do not think the current code is at all clear, mainly because of the use of an unnamed constant 128. I think this code will be much simpler we do a standard `char` comparison.

Reply:
It seems like `c is >= 'a' and <= 'z'` is checking something different than `c < 128`. It might be worth digging a bit to see if constant 128 is anywhere in our character encoding helpers, if you haven't already. Otherwise I think a local variable or function `isAscii` with a brief comment like "128 is the top of the ascii range" would be fine.